libgalois/include/katana/Properties.h

```diff
@@ -385,23 +475,33 @@ struct LargeStringReadOnlyProperty {
   using ViewType = StringPropertyReadOnlyView<arrow::LargeStringArray>;
 };
 
-template <typename Props>
-Result<std::shared_ptr<arrow::Table>>
-AllocateTable(uint64_t num_rows, const std::vector<std::string>& names) {
-  constexpr auto num_tuple_elem = std::tuple_size<Props>::value;
-  static_assert(num_tuple_elem != 0);
-  std::shared_ptr<arrow::Table> table;
-  std::vector<katana::PropertyArrowTuple<Props>> rows(num_rows);
-  KATANA_LOG_ASSERT(names.size() == num_tuple_elem);
-  // TODO(gill): Replace this with NUMA allocated buffers.
-  if (auto r = arrow::stl::TableFromTupleRange(
-          arrow::default_memory_pool(), std::move(rows), names, &table);
-      !r.ok()) {
-    KATANA_LOG_DEBUG("arrow error: {}", r);
-    return Result<std::shared_ptr<arrow::Table>>(katana::ErrorCode::ArrowError);
+template <typename T>
+struct StructProperty {
+  using ArrowType = arrow::FixedSizeBinaryType;
+  using ViewType = katana::PODPropertyView<T>;
+
+  static katana::Result<std::shared_ptr<arrow::Table>> Allocate(
+      size_t num_rows, const std::string& name) {
+    auto res = arrow::FixedSizeBinaryType::Make(sizeof(T));
+    if (!res.ok()) {
+      return KATANA_ERROR(katana::ErrorCode::ArrowError, "{}", res.status());
+    }
+
+    auto type = res.ValueOrDie();
+    arrow::FixedSizeBinaryBuilder builder(type);
+    std::vector<uint8_t> data(sizeof(T) * num_rows);
```

Comment:
Use `PODResizeableArray` instead. It will save time in constructing the elements.

Reply:
Most efficient option is `auto st = builder.AppendEmptyValues(num_rows);` (still initializes to 0)

Reply:
Thanks, Daniel. Your option is better. It gets initialization at the same cost of copy.

Hochan, please use this in `AllocateTable` as well.

Reply:
I guess you are talking about `Allocate()`, Roshan?

Reply:
Yes

Reply:
@danielmawhirter , there is no `AppendEmptyValues()` on the repository. Compilation failed due to undefined symbol. Am I missing something?

Reply:
Yes, it is added to version 3.0. I am using 2.0 and am not sure about compatibility. For now, I am gonna use `PODResizeableArray` and left todo.